tensorly/decomposition/tests/test_tucker.py

```diff
@@ -93,14 +90,12 @@ def test_tucker(monkeypatch):
         assert_equal(
             factors[i].shape,
             (tensor.shape[i], ranks[i]),
-            err_msg="factors[{}].shape={}, expected {}".format(
-                i, factors[i].shape, (tensor.shape[i], ranks[i])
-            ),
+            err_msg=f"{factors[i].shape = }, expected {(tensor.shape[i], ranks[i])}"
```

Comment:
Changed to support Python 3.7

```suggestion
            err_msg=f"factors[i].shape = {factors[i].shape}, expected {(tensor.shape[i], ranks[i])}"
```